nixos/lib/eval-config-standard.nix

```
@@ -0,0 +1,68 @@
+# DO NOT IMPORT. Use nixpkgsFlake.lib.nixos, or import (nixpkgs + "/nixos/lib")
+{ lib }: # read -^
+
+let
+
+  # Base: Use the miminal module evaluation
+  evalModulesMinimal = (import ./default.nix {
+    inherit lib;
+    # Implicit use of feature is noted in implementation.
+    featureFlags.minimalModules = { };
+  }).evalModules;
+
+  baseModules = import ../modules/module-list.nix;
+
+  /*
+    Invoke NixOS. This is the official entrypoint for evaluating a standard
+    NixOS. All additional configuration or implementations has to be passed
+    via the module system as the `modules` parameter or imported using
+    `imports` in modules.
+
+    An official entrypoint clarifies the public interface of a NixOS
+    evaluation and allows the ecosystem to converge.
+
+    Parameters:
+
+      modules:        A list of modules that constitute the configuration.
+
+      specialArgs:    An attribute set of module arguments. Unlike
+                      `config._module.args`, these are available for use in
+                      `imports`.
+                      `config._module.args` should be preferred when possible.
+
+    Return:
+
+      An attribute set containing `config.system.build.toplevel` among other
+      attributes. See `lib.evalModules` in the Nixpkgs library.
+
+   */
```

This should be documented in the manual instead.
Comments are for implementation concerns, not for interfaces.
(Unless, we'd run the tooling that nixpkgs/lib uses. NixOS is more picky about tooling though, because it runs as part of each configuration by default. So I don't know if doing that is feasible. Let's keep it simple for this PR.)

```suggestion
  # See nixos/doc/.../....md
  # or https://nixos.org/manual/nixos/unstable/...
  # NOTE: We keep this interface simple, because almost everything can and should be done through `modules`!
```

---

@fricklerhandwerk `./nixos/doc/manual/development` would be ok here?

---

No, `development` isn't really a fit. There isn't a section like "inner workings" or so. The entire development section is geared towards getting people started to write modules.

---

Suggestively addressed in the PR, now.
To make the doc story work, the public testing interface also needs a section.
I added a placeholder and merging this PR would probably (forcibly) create a documentation dependency on the testing lib.

Otherwise, there won't be a spot in the docs where to anchor this.

---

I hope the choice of the documentation structure is ok, the actual doc text needs proper reviewing, though, still.